monoio/src/io/async_write_rent.rs

```diff
@@ -23,8 +23,8 @@ pub trait AsyncWriteRent {
 /// AsyncWriteRentAt: async write with a ownership of a buffer and a position
 pub trait AsyncWriteRentAt {
     /// Write buf at given offset
-    fn write_at<T: IoBufMut>(
-        &self,
+    fn write_at<T: IoBuf>(
+        &mut self,
```

Comment:
I believe `write_at` doesn't need to take `&mut self`.

Reply:
I discussed with @ethe yesterday about this issue. I think we can provide 2 implementation: one is direct impl for the struct, this can take &self; another one is impl trait, this need to take &mut self for others to impl the trait based on it. How do you think?

Reply:
in poll version of read trait, such as `tokio::io::AsyncRead`, it takes self as `Pin<&mut Self>`, this allows inner mutations, such as buffered reading, and also `monoio::io::AsyncReadRent` takes `&mut self` in the same reason.

Reply:
Make sense to me.